simulators/discovery/version4/discv4.go

```diff
@@ -0,0 +1,151 @@
+package main
```

Comment:
I think this should be in simulators/devp2p. The eth test should also be there, but we can move it some other time.